vllm/distributed/device_communicators/shm_broadcast.py

```diff
@@ -125,8 +125,8 @@ def __init__(self,
                        lambda *args, **kwargs: None):
                 try:
                     self.shared_memory = shared_memory.SharedMemory(name=name)
-                    assert (
-                        self.shared_memory.size == self.total_bytes_of_buffer)
```

Comment:
why do we need this?

Reply:
On macOS, SHM is divided according to the memory page size. If the size is an integer multiple of the page size, an integer multiple of the page memory is allocated, so the shm size will be larger than the passed size.

See: https://docs.python.org/3.13/library/multiprocessing.shared_memory.html#multiprocessing.shared_memory.SharedMemory

> size ([int](https://docs.python.org/3.13/library/functions.html#int)) – The requested number of bytes when creating a new shared memory block. Because some platforms choose to allocate chunks of memory based upon that platform’s memory page size, the exact size of the shared memory block may be larger or equal to the size requested. When attaching to an existing shared memory block, the size parameter is ignored.

Reply:
Do I need to add a comment for this?

Reply:
yes please comment it.

Reply:
Added, PTAL again.